lib/game.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require_relative 'board.rb'
+require_relative 'player.rb'
+require_relative 'ui.rb'
+
+class Game
+  attr_accessor :continue
+  def initialize(board, player1, player2, ui)
+    @new_board = board
+    @player1 = player1
+    @player2 = player2
+    @ui = ui
+    @ui.show_board(@new_board)
+    play
+  end
+
+  def play
+    @player1.assign_sign(@ui)
+    @player2.sign = @player1.sign == 'X' ? 'O' : 'X'
+    while @new_board.check_board
+      cell = @ui.choose_num(@new_board)
+      @player1.make_move(@new_board, cell.to_i)
+      @ui.show_board(@new_board)
+      is_winner = check_game
+      if is_winner == true
```

Comment:
Likewise, lines 26-32 are essentially doing the same thing as lines 38-44. So, it will ease the burden of `play` method to subcontract this to another (most probably `private`) method.